tide_authenticated_content.module

```diff
@@ -353,3 +354,28 @@ function tide_authenticated_content_entity_view_alter(array &$build, EntityInter
     }
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function tide_authenticated_content_form_user_login_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  array_unshift($form['#validate'], '_tide_authenticated_content_user_role_validation');
+}
+
+/**
+ * User Roles can be blocked from logging in to Drupal.
+ */
+function _tide_authenticated_content_user_role_validation($form, FormStateInterface $form_state) {
+  $result = \Drupal::service('user.auth')
+    ->authenticate($form_state->getValue('name'), $form_state->getValue('pass'));
```

Comment:
Hi @anthony-malkoun , thanks for your CR.
IMO, I think we are altering the log in process. we have to run `\Drupal::service('user.auth')->authenticate()`, otherwise, we don't know what does the user input. 
  - `authenticate(string $username, string $password)` does two things
       1. search account name
       2. check if input password hash equals to user stored password hash.
  -  please see https://github.com/dpc-sdp/tide_authenticated_content/pull/5/files#diff-9d70262eca695b61a739997b50337b73R362 and it will be running firstly. no other hooks or validations run before `_tide_authenticated_content_user_role_validation`
Thoughts?

Reply:
This just concerns me. For example if they are a valid front end user, does this update last successful login if `authenticate` passes? Can't we just load the user based on `$username` and get the roles from it then?

Reply:
Hi @anthony-malkoun ,Thanks for your comment.
Yes, we can do it without using `\Drupal::service('user.auth')->authenticate()` please see my updates.